src/main/java/de/bonndan/nivio/output/map/hex/Hex.java

```diff
@@ -65,6 +68,10 @@ public Hex(int q, int r, int s) {
         this.s = s;
     }
 
+    public Hex(int q, int r) {
+        this(q, r, (r + q) * -1);
+    }
+
```

Comment:
Where does this magic formula come from? :)

Reply:
https://www.redblobgames.com/grids/hexagons/implementation.html

TLDR: q and r and like x and y and sufficient to describe a hex position. s is a third axis orthogonal to q and r.

Reply:
I am trying to really understand this. So I understood that it comes from cube algorithms, and thus s would equal z in a cartesian coordinate system.  

The comment helps to understand that the formula has something to do with a coordinate system, but it does not explain why z = (x + y) * (-1)

In general all these Hex and Layout classes are really hard to understand, without understanding the math and logic behind it. 

Initially we used the linked implementation as a library but then started to implement own logic on top of it, right? 

Having own complex code, without (me) really understanding what it does makes it hard to review and grow it. Isn't there a possibility to use the library as it was and decorate (or maybe another pattern is better suited) with our own logic? That own logic can be well tested and understandable :)

Reply:
No, the hex stuff has never been in a library. That was the layouting (force based positioning stuff). Some time ago I investigated some Java hex libraries, but decided against them. If we find a suitable one, we can consider replacing some of our code.

Reply:
Btw, I understood the "s" coordinate as an alternative coordinate, and not as "z". Think of a square with the two sides running parallel to x and y axes in a cartesian system: the sides can be described by offsetting the axes. A hex system has three axes, but you only need to know the offsets of two axes to construct a hex at a position.